smartdashboard/schemas/orchestrator.py

```diff
@@ -24,23 +24,31 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-import pytest
-
-from smartdashboard.utils.helpers import get_db_hosts
-
-from ..utils.test_entities import *
-
-
-@pytest.mark.parametrize(
-    "orchestrator, expected_value",
-    [
-        pytest.param(orchestrator_1, ["shard1_host", "shard2_host"]),
-        pytest.param(orchestrator_2, ["shard1_host", "shard2_host"]),
-        pytest.param(orchestrator_3, ["shard1_host"]),
-        pytest.param(application_1, []),
-        pytest.param(no_shards_orchestrator, []),
-    ],
-)
-def test_get_db_hosts(orchestrator, expected_value):
-    hosts = get_db_hosts(orchestrator)
-    assert hosts == expected_value
+import typing as t
+
+from pydantic import field_validator
+
+from smartdashboard.schemas.base import BaseEntity
+from smartdashboard.schemas.shard import Shard
+
+
+class Orchestrator(BaseEntity):
+    type: str
+    interface: t.List[str] = []
+    shards: t.List[Shard] = []
+
+    @field_validator("interface", mode="before")
+    @classmethod
+    def convert_interface(cls, value: t.Union[str, t.List[str]]) -> t.List[str]:
+        if isinstance(value, str):
+            return [value]
+
+        return value
+
+    @property
+    def ports(self) -> t.Sequence[int]:
+        return tuple({shard.port for shard in self.shards})
```

Comment:
isn't `{shard.port for shard in self.shards}` already a tuple comprehension without wrapping w/`tuple(...)`? I think you might be able to get rid of an extra copy op on the result of the comprehension

Reply:
derp. now i see. you're using the set to filter dupes, then tuple-fying. ignore me.